### PR TITLE
Upgrade kustomization files to Kustomize v5

### PR DESCRIPTION
**Pull request by oksanabaza**
## Why are these changes needed?

This PR addresses the migration from Kustomize v4 to v5. Kuberay was generating warnings about deprecated parameters, which have now been updated to align with Kustomize v5 standards.
## Checks

- [ ] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by astefanutti** [comment]
@kevin85421 could you please have a quick look?

**Comment by kevin85421** [comment]
> @kevin85421 could you please have a quick look?

Sure, sorry I didn’t know it was an urgent item for you all. I initially planned to review it after Ray Summit.

**Comment by kevin85421** [comment]
@oksanabaza would you mind rebasing with the master branch? Thanks!

**Review by kevin85421 — COMMENTED** [review]
I manually test deploying KubeRay operator by running:

1. `kubectl create -k ray-operator/config/default`
2. `make deploy-with-webhooks`

It works. @oksanabaza would you mind manually testing deploying KubeRay apiserver?

**Comment by kevin85421** [comment]
> It works. @oksanabaza would you mind manually testing deploying KubeRay apiserver?

Could you provide some details in the PR description to explain how you tested deploying the KubeRay API server? Thanks!

**Comment by oksanabaza** [comment]
Hey @kevin85421 , I deployed using the following commands:

1. `kubectl create -k ray-operator/config/default`
2. `make deploy-with-webhooks`

Everything worked as expected

Also here are the steps I followed for manual testing:

- Ran kustomize build . > original-v4.yaml on the original v4 manifests
- Updated the manifests to use Kustomize v5 labels
- Ran kustomize build . > updated-v5.yaml on the updated manifests
- Compared the outputs using diff original-v4.yaml updated-v5.yaml
- Verified that there were no differences and no warnings after the Kustomize build commands

Let me know if you need any more information

**Comment by kevin85421** [comment]
cc @MortalHappiness could you also review this PR if you have time?

File: apiserver/deploy/base/insecure/kustomization.yaml [hunk 1]
```diff
@@ -6,9 +6,10 @@ namespace: ray-system
 resources:
 - apiserver.yaml
 
-commonLabels:
-  app.kubernetes.io/name: kuberay
-  app.kubernetes.io/component: kuberay-apiserver
+labels:
```
Comment (MortalHappiness) [inline apiserver/deploy/base/insecure/kustomization.yaml:9]: Add `includeSelectors: true`

File: apiserver/deploy/base/insecure/kustomization.yaml [hunk 2]
```diff
@@ -6,9 +6,10 @@ namespace: ray-system
...
+  - pairs:
+      app.kubernetes.io/name: kuberay
+      app.kubernetes.io/part-of: kuberay-apiserver
```
Comment (MortalHappiness) [inline apiserver/deploy/base/insecure/kustomization.yaml:12]: Why this is changed from `component` to `part-of`?

File: apiserver/deploy/base/secure/kustomization.yaml [hunk 3]
```diff
@@ -6,9 +6,10 @@ namespace: ray-system
 resources:
 - apiserver.yaml
 
-commonLabels:
-  app.kubernetes.io/name: kuberay
-  app.kubernetes.io/component: kuberay-apiserver
+labels:
+  - pairs:
+      app.kubernetes.io/name: kuberay
+      app.kubernetes.io/part-of: kuberay-apiserver
```
Comment (MortalHappiness) [inline apiserver/deploy/base/secure/kustomization.yaml:12]: ditto, all `labels` fields have the same problem as the above.

File: ray-operator/config/manager/kustomization.yaml [hunk 4]
```diff
@@ -4,6 +4,7 @@ resources:
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
-commonLabels:
-  app.kubernetes.io/component: kuberay-operator
-  app.kubernetes.io/name: kuberay
+labels:
+  - pairs:
+      app.kubernetes.io/name: kuberay-operator
+      app.kubernetes.io/part-of: kuberay
```
Comment (MortalHappiness) [inline ray-operator/config/manager/kustomization.yaml:10]: ditto

File: ray-operator/config/rbac/kustomization.yaml [hunk 5]
```diff
@@ -5,6 +5,7 @@ resources:
 - leader_election_role_binding.yaml
 - service_account.yaml
 
-commonLabels:
-  app.kubernetes.io/name: kuberay
-  app.kubernetes.io/component: kuberay-operator
+labels:
+  - pairs:
+      app.kubernetes.io/name: kuberay
+      app.kubernetes.io/part-of: kuberay-operator
```
Comment (MortalHappiness) [inline ray-operator/config/rbac/kustomization.yaml:11]: ditto

File: ray-operator/config/rbac/kustomization.yaml [hunk 6]
```diff
@@ -5,6 +5,7 @@ resources:
 - leader_election_role_binding.yaml
 - service_account.yaml
```
Comment (MortalHappiness) [inline ray-operator/config/rbac/kustomization.yaml:7]: Add

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
```

File: ray-operator/config/default/kustomization.yaml [hunk 7]
```diff
@@ -9,10 +9,10 @@ namespace: ray-system
 #namePrefix: ray-operator-
 
 # Labels to add to all resources and selectors.
-#commonLabels:
+#labels:
 #  someName: someValue
```
Comment (MortalHappiness) [inline ray-operator/config/default/kustomization.yaml:14]: ditto

**Review by MortalHappiness — COMMENTED** [review]
I ran `kustomize edit fix` on all old files and find these inconsistencies with your fix. Do you have any specific reasons not to use the `kustomize edit fix` command directly? If yes, could you give some reasons why they should be different?

**Comment by kevin85421** [comment]
@oksanabaza is this PR ready for another review?

File: apiserver/deploy/base/insecure/kustomization.yaml (hunk 1 above)
Comment (oksanabaza) [reply apiserver/deploy/base/insecure/kustomization.yaml]: thanks, have added `includeSelectors: true`

File: apiserver/deploy/base/insecure/kustomization.yaml (hunk 2 above)
Comment (oksanabaza) [reply apiserver/deploy/base/insecure/kustomization.yaml]: thanks, changed to `component`

File: apiserver/deploy/base/secure/kustomization.yaml (hunk 3 above)
Comment (oksanabaza) [reply apiserver/deploy/base/secure/kustomization.yaml]: this one also fixed, thanks

File: ray-operator/config/manager/kustomization.yaml (hunk 4 above)
Comment (oksanabaza) [reply ray-operator/config/manager/kustomization.yaml]: same here

File: ray-operator/config/rbac/kustomization.yaml (hunk 5 above)
Comment (oksanabaza) [reply ray-operator/config/rbac/kustomization.yaml]: same here

File: ray-operator/config/rbac/kustomization.yaml (hunk 6 above)
Comment (oksanabaza) [reply ray-operator/config/rbac/kustomization.yaml:7]: thanks, added

File: ray-operator/config/default/kustomization.yaml (hunk 7 above)
Comment (oksanabaza) [reply ray-operator/config/default/kustomization.yaml:14]: @MortalHappiness thanks for the review, there were no specific reasons not to use the `kustomize edit fix` command directly so after the review I used it to fix those deprecated values/ fix missing fields

**Comment by oksanabaza** [comment]
@kevin85421 @MortalHappiness could please review it again, thanks!

**Review by MortalHappiness — APPROVED** [review]
LGTM

**Review by MortalHappiness — CHANGES_REQUESTED** [review]
I found that some files are not changed. Please run `kustomize edit fix` for all of these files. For example, you didn't fix `ray-operator/config/certmanager/kustomization.yaml` in this PR.

```
apiserver/deploy/base/insecure/kustomization.yaml
apiserver/deploy/base/secure/kustomization.yaml
apiserver/deploy/local/insecure/kustomization.yaml
apiserver/deploy/local/secure/kustomization.yaml
ray-operator/config/certmanager/kustomization.yaml
ray-operator/config/crd/kustomization.yaml
ray-operator/config/default/kustomization.yaml
ray-operator/config/default-with-webhooks/kustomization.yaml
ray-operator/config/manager/kustomization.yaml
ray-operator/config/rbac/kustomization.yaml
ray-operator/config/webhook/kustomization.yaml
```

**Comment by oksanabaza** [comment]
@MortalHappiness thank you for the comment. I've run `kustomize edit fix` for all of them. Could you please check it again? Thank you

**Review by MortalHappiness — APPROVED** [review]
LGTM

**Review by kevin85421 — APPROVED** [review]
Thanks!

**Pull request merged**